robottelo/datafactory.py

```diff
@@ -428,3 +429,33 @@ def invalid_http_credentials(url_encoded=False):
                 for cred in credentials]
     else:
         return credentials
+
+
+@datacheck
+def get_valid_preUpgrade_data(testEntity, field):
+    """Returns valid list of tuples for given testEntity and field from
+    preUpgrade YAML file
+
+    :param str testEntity: The testEntity name in YAML file
+    e.g. Organization-Tests to check it exists and its associations
+    :param str field: Field name in YAML for which the test created
+    e.g. name to check if the testEntity exists by name
+    e.g. smart-proxy to check association with testEntity
+    :return: List of tuples. Tuple containing pair of SubEntity, field_value
+    e.g. [('Default Organization',2),('another_org',5)], where 2 and 5 are
+    ids(field) of two mentioned orgs(subEntity)
+    """
+    datalist = []
+    for entity in get_all_yaml_data()[testEntity].keys():
```

Comment:
Actually, This is intentional as if the latest additions will be fetched for every run and not for every test, so that we can have latest fresh data. If you run the whole test module it will be downloaded only once.